packages/sui-types/src/lib.rs

```diff
@@ -32,6 +40,12 @@ impl FromStr for SuiAddress {
     type Err = Report<Error>;
 
     fn from_str(s: &str) -> Result<Self, Error> {
+        // disallow uppercase characters for the sui addresses
+        ensure!(
```

Comment:
Add unit test for failing on correctly hex encoded but upper and mixed case strings

Reply:
It's in the `address.rs` tests